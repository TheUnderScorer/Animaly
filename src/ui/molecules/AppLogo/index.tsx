import { RowView } from '../../../styles/view';
import { Text } from '@ui-kitten/components';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
import React, { FC, useMemo } from 'react';
import { StyleSheet } from 'react-native';
import config from '../../../config';

const styles = StyleSheet.create({
  row: {
    justifyContent: 'center',
    alignItems: 'center',
  },
  subtitle: {
    marginTop: 5,
  },
});

export interface AppLogoProps {
  variant?: 'light' | 'dark';
}

const AppLogo: FC<AppLogoProps> = ({ variant }) => {
  const textColor = useMemo(() => (variant === 'dark' ? '#000' : '#fff'), [
    variant,
  ]);

  return (
    <>
      <RowView style={styles.row}>
        <Text
          style={{
            color: textColor,
          }}
          category="h1">
          {config.appName}
        </Text>
        <Icon
          size={40}
          style={{
            color: textColor,
          }}
          name="dog-side"
        />
      </RowView>
      <RowView style={[styles.row, styles.subtitle]}>
        <Text
          style={{
            color: textColor,
          }}
          category="s2">
          Your everyday companion.
        </Text>
      </RowView>
    </>
  );
};

export default AppLogo;
