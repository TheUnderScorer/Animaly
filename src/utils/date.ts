import moment from 'moment';

export type TimeOfDay = 'night' | 'morning' | 'afternoon' | 'evening';

const timesOfDay = {
  '0-5': 'night',
  '6-10': 'morning',
  '11-15': 'afternoon',
  '16-23': 'evening',
};
const timesOfDayArr = Object.entries(timesOfDay);

export const getTimeOfDayText = (date = moment()): TimeOfDay | null => {
  const hour = parseInt(date.format('H'), 10);

  const foundTime = timesOfDayArr.find(([period]) => {
    const [startTime, endTime] = period
      .split('-')
      .map((item) => parseInt(item, 10));

    return hour >= startTime && hour <= endTime;
  });

  return foundTime ? (foundTime[1] as TimeOfDay) : null;
};
