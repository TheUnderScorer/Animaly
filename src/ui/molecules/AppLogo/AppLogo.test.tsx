import { render } from 'react-native-testing-library';
import AppLogo from './';
import React from 'react';
import { wrapComponent } from '../../../tests/utils';

describe('<AppLogo />', () => {
  it('should render without crashing', () => {
    const cmp = render(wrapComponent(<AppLogo />));
    expect(cmp.toJSON()).toMatchSnapshot();
  });
});
