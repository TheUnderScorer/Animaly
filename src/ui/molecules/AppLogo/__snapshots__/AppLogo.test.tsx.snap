// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppLogo /> should render without crashing 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
        },
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
      ]
    }
  >
    <Text
      appearance="default"
      category="h1"
      style={
        Array [
          Object {
            "color": "#222B45",
            "fontFamily": "System",
            "fontSize": 36,
            "fontWeight": "800",
          },
          Object {
            "color": "#fff",
          },
        ]
      }
    >
      Animaly
    </Text>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 40,
          },
          Object {
            "color": "#fff",
          },
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      ï©ƒ
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
        },
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
        Object {
          "marginTop": 5,
        },
      ]
    }
  >
    <Text
      appearance="default"
      category="s2"
      style={
        Array [
          Object {
            "color": "#222B45",
            "fontFamily": "System",
            "fontSize": 13,
            "fontWeight": "600",
          },
          Object {
            "color": "#fff",
          },
        ]
      }
    >
      Your everyday companion.
    </Text>
  </View>
</View>
`;
